mantidimaging/main.py

```diff
@@ -102,26 +102,49 @@ def main() -> None:
 
     q_application = setup_application()
 
+    # Check for non-clean shutdown
+    was_clean = settings.value("app/last_shutdown_clean", True, type=bool)
+    if not was_clean:
+        log_dir = settings.value("logging/log_dir", "")
+        latest_log = None
+        if log_dir:
+            logs = sorted(Path(log_dir).glob("mantid_imaging_*.log"), key=lambda p: p.stat().st_mtime, reverse=True)
+            if logs:
+                latest_log = logs[0]
+
+        msg = QMessageBox()
+        msg.setIcon(QMessageBox.Warning)
+        msg.setWindowTitle("Previous Shutdown Warning")
+        if latest_log and latest_log.exists():
+            msg.setText(f"Mantid Imaging did not shut down cleanly last time.\n\nLog file:\n{latest_log}")
+        else:
+            msg.setText("Mantid Imaging did not shut down cleanly last time.\n\nNo log file is available.")
+        msg.setStandardButtons(QMessageBox.Ok)
+        msg.exec_()
```

Comment:
Can this all be put into a function.

Also the second message could say add "To record log files to disk, configure a directory in File -> Settings"